wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextDsl.kt

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.wrapper.minecraft.text
+
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.server.command.ServerCommandSource
+import net.minecraft.text.MutableText
+import net.minecraft.text.Text
+import net.minecraft.text.data.TextData
+import net.minecraft.util.Identifier
+import java.util.Optional
+import java.util.stream.Stream
+
+/**
+ * DSL Marker for text based DSL's.
+ */
+@DslMarker
+public annotation class TextDslMarker
+
+/**
+ * This class contains the functions for building a [Text] object.
+ * To use the DSL use [buildText].
+ *
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class TextDsl internal constructor(action: TextDsl.() -> Unit) {
+    init {
+        apply(action)
+    }
+
+    /**
+     * The text object being built.
+     *
+     * @author NoComment1105
+     */
+    @PublishedApi
+    internal val text: MutableText = Text.empty()
+
+    /**
+     * Adds a translatable text.
+     *
+     * @param value The translation key of the text.
+     * @param args Any arguments to apply to the translatable text. Can be left
+     * empty for no arguments.
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun translatable(
+        value: String,
+        vararg args: Any,
+        action: MutableStyle.() -> Unit = { }
```

Comment:
The name `action` is weird. One may have this:

```kotlin
translatable(
    value = /* ... */,
    action = {
        // Oh is this for click event?
    }
)
```

I'd prefer `style`.

Reply:
For consistency, we have any lambda parameters use the name `action`

Reply:
> For consistency, we have any lambda parameters use the name `action`

I think it's too fixed under some circumstance and doesn't convey any useful information which is bad.